app/scripts/metamask-controller.js

```diff
@@ -259,6 +259,7 @@ export default class MetamaskController extends EventEmitter {
         MESSAGE_TYPE.ETH_SIGN,
         MESSAGE_TYPE.PERSONAL_SIGN,
         MESSAGE_TYPE.ETH_SIGN_TYPED_DATA,
+        MESSAGE_TYPE.TRANSACTION,
```

Comment:
This will clash with the other pending PR, but we have the `ApprovalType` enum now instead.